index.js

```diff
@@ -1,6 +1,131 @@
+     */
+    this.isGlimmer = emberSourceVersion || (emberBowerChecker.gt('2.9.0-alpha') && emberBowerChecker.lt('2.9.0')) || emberBowerChecker.gt('2.10.0-alpha');
+    this.isHTMLBars = !this.isGlimmer && emberBowerChecker.gt('1.10.0-beta');
+
+    if (this.isGlimmer) {
+      this.versionSpecificPath = path.join(this.root, 'glimmer-version');
+    } else if (this.isHTMLBars) {
+      this.versionSpecificPath = path.join(this.root, 'htmlbars-version');
+    } else {
+      throw new Error('ember-a11y does not support your version of Ember.');
+    }
+  },
+
+  shouldIncludeChildAddon: function(addon) {
+    if (addon.name !== 'ember-getowner-polyfill') {
+      return this._super.shouldIncludeChildAddon.apply(this, arguments);
+    }
+
+    // Only dealing with `ember-getowner-polyfill`
+    var versionChecker = new VersionChecker(this);
+
+    var emberSourceVersion = versionChecker.for('ember-source', 'npm').version;
+    var emberBowerChecker = versionChecker.for('ember', 'bower');
+
+    this.needsOwnerPolyfill = (!emberSourceVersion && emberBowerChecker.lt('2.3.0-beta'));
+    return this.needsOwnerPolyfill;
+  },
+
+  treeForAddon: function() {
+    var tree = mergeTrees([this._super.treeForAddon.apply(this, arguments)], { overwrite: true });
+    var trees = [tree];
+
+    // include `ember-internals` module ONLY for htmlbars
+    if (this.isHTMLBars) {
+      trees.push(new Funnel(this.versionSpecificPath, {
+        files: ['ember-internals.js'],
+        destDir: 'modules/' + this.name
+      }));
+    }
+
+    if (this.needsOwnerPolyfill) {
+      trees.push(replace(tree, {
+        files: ['modules/ember-a11y/mixins/focusing.js'],
+        pattern: {
+          match: /\/\/ import getOwner from 'ember-getowner-polyfill';/,
+          replacement: 'import getOwner from \'ember-getowner-polyfill\';'
+        }
+      }));
+    } else {
+      trees.push(replace(tree, {
+        files: ['modules/ember-a11y/mixins/focusing.js'],
+        pattern: {
+          match: /\/\/ const { getOwner } = Ember;/,
+          replacement: 'const { getOwner } = Ember;'
+        }
+      }));
+    }
+
+    return mergeTrees(trees, { overwrite: true });
+  },
+
+  treeForTemplates: function() {
+    var trees = [this._super.treeForTemplates.apply(this, arguments)];
+
+    trees.push(this.treeGenerator(
+      path.resolve(this.root, this.versionSpecificPath, 'app', 'templates')
+    ));
+
+    return mergeTrees(trees.filter(Boolean), { overwrite: true });
+  },
+
+  treeForApp: function(defaultTree) {
+    this._super.treeForApp.call(this, arguments);
+
+    var PATTERN = /{{htmlbars}}[\s\S]*?{{\/htmlbars}}/gim;
+
+    if (this.isGlimmer) {
+      // since `ember-internals` module is not needed for Glimmer
+      // we want to remove calls to it from instance initializer.
+      // string-replace hack, replace everything that is wrapped
+      // into "{{htmlbars}}" with an empty string.
+      defaultTree = replace(defaultTree, {
```

Comment:
I'm pretty sure it will only change if that file changes: https://github.com/rwjblue/broccoli-string-replace/blob/master/index.js#L1

(As I link you to your own project.)